gammapy/modeling/models/core.py

```diff
@@ -437,17 +438,21 @@ class DatasetModels(collections.abc.Sequence, CovarianceMixin):
     ----------
     models : `SkyModel`, list of `SkyModel` or `Models`
         Sky models.
-    covariance_data : `~numpy.ndarray`
-        Covariance data.
+    covariance_data : `~numpy.ndarray`, optional
+        Covariance data. Default is None.
+    penalties : list of `~gammapy.stats.FitStatisticPenalty`, optional
+        Penalties to be applied to the Models parameters when computing a FitStatistic. Default is None.
     """
 
-    def __init__(self, models=None, covariance_data=None):
+    def __init__(self, models=None, covariance_data=None, penalties=None):
         if models is None:
             models = []
 
         if isinstance(models, (Models, DatasetModels)):
             if covariance_data is None and models.covariance is not None:
                 covariance_data = models.covariance.data
+            if penalties is None and models._penalties is not None:
+                penalties = models._penalties
```

Comment:
In general, there are several penalities for all models. But, as a penalty is computed from a list of parameters, how do you check that 2 penalties are not using the same parameter, ie they are independant?

Reply:
and if parameters are linked, does the penalty is applied once or twice ?